service/cert-signer/cert_signer.go

```diff
@@ -101,6 +128,115 @@ func (cs *certSigner) Issue(config spec.IssueConfig) (spec.IssueResponse, error)
 	return newIssueResponse, nil
 }
 
-func (cs *certSigner) SignedPath(clusterID string) string {
-	return fmt.Sprintf("pki-%s/issue/role-%s", clusterID, clusterID)
+func (cs *certSigner) SignedPath(clusterID string, organizations string) string {
+	return fmt.Sprintf("pki-%s/issue/%s", clusterID, roleName(clusterID, organizations))
+}
+
+func (cs *certSigner) IsRoleCreated(clusterID string, organizations string) (bool, error) {
```

Comment:
By the way

I found out that vault doesn't care if the role already exists before you attempt to create it. Why do we bother to check?

Creating a role is in this sense a idempotent operation. If it already exists, nothing goes wrong. If it doesn't exist, it makes it.